03_Modules/OcppRouter/src/module/webhook.dispatcher.ts

```diff
@@ -0,0 +1,301 @@
+        );
+      }
+    }
+
+    this._onConnectionCallbacks.set(
+      connectionIdentifier,
+      onConnectionCallbacks,
+    );
+    this._onCloseCallbacks.set(connectionIdentifier, onCloseCallbacks);
+    this._onMessageCallbacks.set(connectionIdentifier, onMessageCallbacks);
+    this._sentMessageCallbacks.set(connectionIdentifier, sentMessageCallbacks);
+  }
+
+  private _onConnectionCallback(subscription: Subscription) {
+    return (info?: Map<string, string>) =>
+      this._subscriptionCallback(
+        {
+          stationId: subscription.stationId,
+          event: 'connected',
+          info: info,
+        },
+        subscription.url,
+      );
+  }
+
+  private _onCloseCallback(subscription: Subscription) {
+    return (info?: Map<string, string>) =>
+      this._subscriptionCallback(
+        { stationId: subscription.stationId, event: 'closed', info: info },
+        subscription.url,
+      );
+  }
+
+  private _onMessageReceivedCallback(subscription: Subscription) {
+    return async (message: string, info?: Map<string, string>) => {
+      if (
+        !subscription.messageRegexFilter ||
+        new RegExp(subscription.messageRegexFilter).test(message)
+      ) {
+        return this._subscriptionCallback(
+          {
+            stationId: subscription.stationId,
+            event: 'message',
+            origin: MessageOrigin.ChargingStation,
+            message: message,
+            info: info,
+          },
+          subscription.url,
+        );
+      } else {
+        // Ignore
+        return true;
+      }
+    };
+  }
+
+  private _onMessageSentCallback(subscription: Subscription) {
+    return async (message: string, error?: any, info?: Map<string, string>) => {
+      if (
+        !subscription.messageRegexFilter ||
+        new RegExp(subscription.messageRegexFilter).test(message)
+      ) {
+        return this._subscriptionCallback(
+          {
+            stationId: subscription.stationId,
+            event: 'message',
+            origin: MessageOrigin.ChargingStationManagementSystem,
+            message: message,
+            error: error,
+            info: info,
+          },
+          subscription.url,
+        );
+      } else {
+        // Ignore
+        return true;
+      }
+    };
+  }
+
+  /**
+   * Sends a message to a given URL that has been subscribed to a station connection event
+   *
+   * @param {Object} requestBody - request body containing stationId, event, origin, message, error, and info
+   * @param {string} url - the URL to fetch data from
+   * @return {Promise<boolean>} a Promise that resolves to a boolean indicating success
+   */
+  private async _subscriptionCallback(
+    requestBody: {
+      stationId: string;
+      event: string;
+      origin?: MessageOrigin;
+      message?: string;
+      error?: any;
+      info?: Map<string, string>;
+    },
+    url: string,
+  ): Promise<boolean> {
+    try {
+      const response = await fetch(url, {
```

Comment:
Can we utilize `typed-rest-client` here. We use this for performing HTTP requests in OCPI for PUSH messages, perhaps it can also be helpful here. We do have a BaseClientApi that may not 100% fit into this because we set OCPI headers and stuff, so perhaps we can incorporate a new, more generic base... It can be good to reuse the same rest client when performing external HTTP requests to keep things organized..